tests/argument_matching_tests.cpp

```diff
@@ -165,6 +167,21 @@ struct ArgumentMatchingTests: tpunit::TestFixture {
 		Verify(Method(mock, func).Using(Ne(10))).Twice();
 	}
 
+	void test_str_eq_matcher() {
+
+		Mock<SomeInterface> mock;
+
+		When(Method(mock, strfunc).Using(StrEq("first"))).Return(1);
+		When(Method(mock, strfunc).Using(StrEq("second"))).Return(2);
+
+		SomeInterface &i = mock.get();
+		ASSERT_EQUAL(1, i.strfunc("first"));
```

Comment:
Will this work if the const char* is pointing to a temporary, instead of a literal?

Reply:
I looked a bit more, Matcher aren't really design to hold copy of stuff, they just hold references. The issue is the same for all matcher, so I won't change it only for this one.

Reply:
Like you said in another message maybe it could be improved depending of the value category of the variable but it's outside of the scope of this PR i think.

Reply:
yeah - I see no problem. Could be a future improvement. (I'm just seeing lots of scope for small improvements in this library in general)

Reply:
Re "references" in general. There are a number of cases where Fakeit stops temporaries and/or rvalue refs  from being used (this one, `const char*`, is less likely to have issues with temporaries though). 
And so sometimes makes the tests a little more complicated than necessarily, and adds more gotchas, where it might be possible to avoid.